src/common/datomish/query.cljc

```diff
@@ -142,15 +162,37 @@
   (let [{:keys [find in with where]} find]  ; Destructure the Datalog query.
     (validate-with with)
     (validate-in in)
+
+    ;; A find spec can be:
+    ;;
+    ;;   * FindRel containing :elements. Returns an array of arrays.
+    ;;   * FindColl containing :element. This is like mapping (fn [row] (aget row 0))
+    ;;     over the result set. Returns an array of homogeneous values.
+    ;;   * FindScalar containing :element. Returns a single value.
+    ;;   * FindTuple containing :elements. This is just like :limit 1
+    ;;     on FindColl, returning the first item of the result array. Returns an
+    ;;     array of heterogeneous values.
+    ;;
+    ;; The code to handle these is:
+    ;;   - Just above, unifying a variable list in find-spec->elements.
+    ;;   - In context.cljc, checking whether a single value or collection is returned.
+    ;;   - In projection.cljc, transducing according to whether a single column or
+    ;;     multiple columns are assembled into the output.
+    ;;   - In db.cljc, where we finally take rows and decide what to push into an
+    ;;     output channel.
+
     (let [external-bindings (in->bindings in)
-          elements (:elements find)
+          elements (find-spec->elements find)
           known-types {}
           group-by-vars (projection/extract-group-by-vars elements with)]
-      (assoc context
-             :elements elements
-             :group-by-vars group-by-vars
-             :has-aggregates? (not (nil? group-by-vars))
-             :cc (clauses/patterns->cc (:default-source context) where known-types external-bindings)))))
+      (util/assoc-if
+        (assoc context
+               :find-spec find
+               :elements elements
+               :group-by-vars group-by-vars
+               :has-aggregates? (not (nil? group-by-vars))
+               :cc (clauses/patterns->cc (:default-source context) where known-types external-bindings))
+        :limit (find-spec->limit find)))))
```

Comment:
If we decide to use `LIMIT 1` because of the find spec, then yes, this will win — `find-into-context` happens after `options-into-context` in `<?q`.

Probably no point being needlessly whips-and-chains about specifying a limit being wrong just because we decide we can use a lower one.